spec/support/mail_matcher.rb

```diff
@@ -13,17 +15,40 @@
   end
 end
 
-RSpec::Matchers.define(:include_in_html_body) do |expected|
-  # ignore \r and \n in the comparison
-  mail_body_stripped = ->(mail) { mail.html_part.body.decoded.gsub(/[\r\n]/, "") }
-  expected_stripped = expected.gsub(/[\r\n]/, "")
+RSpec::Matchers.define(:include_in_html_body) do |expected_text|
+  @body_doc = nil
 
   match do |mail|
-    mail_body_stripped.call(mail).include?(expected_stripped)
+    return false unless mail&.html_part&.body
+
+    # Parse the document once and store the Nokogiri object.
+    # The .to_s is kept for clarity and safety, ensuring a string is passed to Nokogiri.
+    @body_doc = Nokogiri::HTML(mail.html_part.body.to_s)
+    body_text = @body_doc.text
+
+    # Normalize whitespace in both strings before comparison.
```

Comment:
Remove comment. It's clear that we deal with normalization since the variable is called `normalized_body`. It's also clear that these lines remove whitespaces by the squish [documentation](https://apidock.com/rails/String/squish) one can retrieve by `ctrl + clicking` on the the word `squish`:

> Returns the string, [first](https://apidock.com/rails/String/first) removing all whitespace on both ends of the string, and then changing remaining consecutive whitespace groups into one space each.